api/base/views.py

```diff
@@ -75,3 +79,21 @@ def root(request, format=None):
             'users': absolute_reverse('users:user-list'),
         }
     })
+
+
+class OsfAPIViewMeta(type):
+    """Metaclass to ensure a base level of permissions are applied to all of our views
+    """
+    def __new__(cls, name, bases, attributes):
+        permission_classes = set((
+            drf_permissions.IsAuthenticatedOrReadOnly,
+            base_permissions.TokenHasScope,
+        ))
+        bases_permission_classes = attributes.get('permission_classes', tuple())
+        try:
+            iter(bases_permission_classes)
+        except TypeError:
+            bases_permission_classes = (bases_permission_classes, )
+        bases_permission_classes = set(bases_permission_classes)
```

Comment:
The goal here was to avoid dealing with instances. 

I'm probably confused, but though attributes would contain the attributes after applying the inheritance chain in bases. If that does not work we can always iterate over the bases tuple to get the base classes permissions classes